File: library/src/main/java/me/proxer/library/enums/Season.java
package me.proxer.library.enums;

import com.squareup.moshi.Json;

/**
 * Enum holding the available seasons.
 *
 * @author Ruben Gees
 */
public enum Season {
    @Json(name = "0") UNSPECIFIED_ALT,
    @Json(name = "1") WINTER,
    @Json(name = "2") SPRING,
    @Json(name = "3") SUMMER,
    @Json(name = "4") AUTUMN,
    @Json(name = "5") UNSPECIFIED
}

File: library/src/main/kotlin/me/proxer/library/enums/Season.kt
package me.proxer.library.enums

import com.serjltt.moshi.adapters.FallbackEnum
import com.squareup.moshi.Json
import com.squareup.moshi.JsonClass

/**
 * Enum holding the available seasons.
 *
 * @author Ruben Gees
 */
@JsonClass(generateAdapter = false)
@FallbackEnum(name = "UNSPECIFIED")
enum class Season {

    @Json(name = "1")
    WINTER,

    @Json(name = "2")
    SPRING,

    @Json(name = "3")
    SUMMER,

    @Json(name = "4")
    AUTUMN,

    @Json(name = "5")
    UNSPECIFIED
}
